src/components/DataControllerMixin/DataControllerMixin.js

```diff
@@ -51,6 +51,16 @@ var ModalMixin = {
         area: props.area || null,
         time: timeidx,
       },
+    }).then(function(data, textstatus, jqXHR) {
+      if($.isEmptyObject(data)) {
+        return $.Deferred().reject(jqXHR, "empty", textstatus);
+      }
+      for(var run in data) {
+        if(!('data' in data[run]) || !('units' in data[run])){
+          return $.Deferred().reject(jqXHR, "incomplete", textstatus);            
+        }
+      }
+      return $.Deferred().resolve(data, textstatus, jqXHR);
```

Comment:
I fear that this, and similar code for the other `getXPromise` functions may not do what you'd like it to do. [This is because jQuery < 3 doesn't actually implement what most of the JS world means by Promise](http://stackoverflow.com/questions/23744612/problems-inherent-to-jquery-deferred-jquery-1-x-2-x). Domenic Denicola wrote an [explanatory article](https://blog.domenic.me/youre-missing-the-point-of-promises/) about this several years ago which is still valid. 

(In my first big outing with jQuery promises about 5 years ago, I tripped all over these issues without understanding them at the time. They're gnarly, not least because the terminology overlaps but means similar but also importantly different things in the two contexts. Since then I've used and come to greatly appreciate Promises/A+ compliant promises, which are now part of the latest JavaScript standard (ES6).)

Here is the problem I believe you have:

1. `$.ajax()` returns a [`jqXHR`](http://api.jquery.com/jQuery.ajax/#jqXHR) object. 
2. A `jqXHR` object implements the [jQuery Promise](http://api.jquery.com/deferred.promise/) interface.
3. The jQuery Promise interface is a subset of the [jQuery Deferred](http://api.jquery.com/category/deferred-object/) interface.
4. The jQuery Deferred/Promise interface only provides for attaching callbacks for various Deferred events. It doesn't return the value (or, more properly a promise for the value) of what the handlers return. This is the issue taken up in the two links above.
5. This callback handler's purpose appears to be to: 
    - transform the result of the response to the Ajax request to a Deferred -- this it does
    - return that Deferred to the caller of the `getDataPromise` function -- this it doesn't, because of how jQuery Deferreds/Promises work
6. Instead, I believe  `getDataPromise` will return the original jqXHR object, and your client code will simply attach another callback to that object, not to the nice Deferreds you manufactured in the callback handler immediately above.

I could be wrong about this -- haven't used jQuery for some while -- but if this code doesn't work, the above is the reason why.